### PR TITLE
Allow `BallistaContext::read_*` methods to read multiple paths.

### LINKED ISSUE #584: Add `read_json` method to ballista
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
I want to read json files from ballista. I can see `read_csv`, `read_avro` and `read_parquet` but not `read_json`. It looks like the method is avilable in datafusion: [Location](https://github.com/apache/arrow-datafusion/blob/24023b5de7cc6c1223ce508d8686d70a8cc8110d/datafusion/core/src/execution/context.rs#L587). Please add this method to ballista.

**Describe the solution you'd like**
Ability to read the json files from ballista using a method similar to `read_csv`.

**Describe alternatives you've considered**

**Additional context**
I want to read a path from S3 which has json files.

### LINKED ISSUE #587: Read multiple files/folders using `read_csv`
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
I want to load files from multiple locations. In Spark, I can just give a list of paths and it works. Would like to do the same with `read_csv`, `read_avro` and `read_parquet` methods.

**Describe the solution you'd like**
Read a list of files using methods like so: `read_csv(vec![String::new()])`

**Describe alternatives you've considered**
I can read the paths one by one and then concatenate the dataframes but not sure if it is ok to do so.

**Additional context**
I have files from S3 in multiple locations and use the above methods to read them.

### PR DESCRIPTION
**Pull request by luckylsk34**
# Which issue does this PR close?
Closes #584. Closes #587.

 # Rationale for this change
Add the ability to read multiple files using `read_*` methods.

# What changes are included in this PR?
1. Added `read_json` method
2. Modified function signature to take any argument implementing `DataFilePaths` trait.

# Are there any user-facing changes?
Yes, they will be able to use `read_*` methods as follows: `read_csv(vec!["", ""])`

### CODE REVIEW

**Comment by luckylsk34** [comment]
@alamb @andygrove pls review.

**Comment by andygrove** [comment]
LGTM. The changes seem consistent with recent changes in DataFusion. It looks like there are some build issues to be resolved. Thanks @luckylsk34

File: ballista-cli/Cargo.toml [hunk 1]
```diff
@@ -33,8 +33,8 @@ ballista = { path = "../ballista/client", version = "0.10.0", features = [
     "standalone",
 ] }
 clap = { version = "3", features = ["derive", "cargo"] }
-datafusion = "18.0.0"
-datafusion-cli = "18.0.0"
+datafusion = { git = "https://github.com/apache/arrow-datafusion" }
```
[the remark this answers is not in the document]
Comment (saikrishna1-bidgely) [reply ballista-cli/Cargo.toml]: added the specific revision.

[the remark this answers is not in the document]
Comment (luckylsk34) [reply ballista-cli/Cargo.toml]: Done, added the specific revision.

**Comment by luckylsk34** [comment]
@andygrove I've added the specific revision and fixed the conflicts, pls approve the workflows. Lets see if we again get the build issues.

**Comment by luckylsk34** [comment]
@andygrove build is a success. Any more changes needed?

**Pull request merged**